encode.js

```diff
@@ -0,0 +1,115 @@
+const { readdirSync, readFileSync, writeFileSync } = require("fs");
+const { parse } = require("yaml");
+
+// get yaml files in this repo
+const files = readdirSync(__dirname).filter((file) => /\.ya?ml$/.test(file));
```

Comment:
I'm not 100% comfortable with parsing all `*.yaml` files; people often use root YAML files for controlling CI/CD systems and adding other metadata. While I get that most users are just going to fork this repo and edit `redirects.yaml`, it's IMHO not impossible that someone would want to annotate all their repos, including private ones, with extra root YAML files. On a side note, I'm curious what the use case is for merging all the YAML files versus just having one specific file that contains the redirects.

I'd prefer either just parsing a specific file, a more specific glob, e.g., `redirects*.ya?ml`, or perhaps allowing the end user to specify the file and/or glob themselves.

Reply:
Allowing multiple redirect files was at the request of Julie, who wanted that because tislab and b2ai have a very large number of links that they want to split up in different ways. Granted, this may be less necessary, now that I have switched from `.json` to `.yaml`, which is both more compact and allows comments that could split up large sections of links.

For now, I've changed it so that there's a dedicated `/redirects` folder that the script reads all the `.yaml` files from. But I may revert this change. I'd prefer to keep things simpler... everything in root as much as possible, right in the user's face, no extra click. I think anyone doing anything else fancy -- like switching from GitHub Actions to Travis where the config yaml has to be in the root folder -- is going to be rare (almost defeats the benefit of using this pre-set-up approach) and the responsibility is on them to modify the code.

@bot2ai What do you think about this?

Reply:
Well, fair enough. Feel free to revert it.

Reply:
I've reverted it back to top-level .yaml files, but I'm now doing it via a glob pattern as you suggested, which is more easily customizable by the user if they edit the script. Also, the pattern I used -- `*.y?(a)ml` -- seems to have the nice benefit that it skips over files starting with `.`, which many/most config files have.

I also tried to rewrite the readme text a bit to be more vague. It only mentions "top folder yaml files" upfront once, and all subsequent times just says "yaml files". So if a user edits the glob, they should only have to change one part of the readme.